### PR TITLE
Add basic auth and security context options to Helm chart

### LINKED ISSUE #237: When using the Helm installer OpenFaaS Gateway doesn't work with Istio
When installing OpenFaaS with Helm on Istio enabled namespaces the Gateway pod enters `Init:CrashLoopBackOff`.

## Expected Behaviour

The Helm chart should offer on option that would allow Istio init container to run as root.

## Current Behaviour

The Gateway pod Istio init container errors out with:

```
iptables v1.6.0: can't initialize iptables table `nat': Permission denied (you must be root)
Perhaps iptables or your kernel needs to be upgraded.
```

## Possible Solution
Disable the Gateway security context.

## Steps to Reproduce (for bugs)

1. Install Istio
2. Add the `istio-injection: true` label to openfaas namepaces
3. Deploy OpenFaaS with Helm

## Your Environment

* GKE 1.10 with Istio 0.8

### PR DESCRIPTION
**Pull request by stefanprodan**
## Description
- add security context option to Helm chart (enabled by default)
- fix #237 when used with Istio the security context must be disabled for Istio init container to run as root
- add basic auth option and document how to enable it
- bump Helm chart version to 1.1.1

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
Security context option was tested on GKE 1.10 with Istio 0.8.
Basic auth option was tested on GKE 1.10 with https://github.com/stefanprodan/openfaas-flux

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
While here should we make the operator only use a Role/RoleBinding Vs ClusterRole or was there something else preventing it? I know Michael G wanted to ask you about this.

File: chart/openfaas/README.md [hunk 1]
```diff
@@ -87,6 +87,27 @@ By default services will be exposed with following hostnames (can be changed, se
 * `prometheus.openfaas.local`
 * `alertmanager.openfaas.local`
 
+### Secure the Gateway administrative API and UI with basic auth
+
+In order to enable basic auth first you need to create a secret named `basic-auth` in the `openfaas` namespace:
+
+```bash
+# generate a random password
+password=$(head -c 12 /dev/random | shasum| cut -d' ' -f1)
```
Comment (alexellis) [inline chart/openfaas/README.md]: I found that this hangs on DigitalOcean, using `/dev/urandom` is faster and should work regardless without blocking.

File: chart/openfaas/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Enable Kubernetes as a backend for OpenFaaS (Functions as a Service)
 name: openfaas
-version: 1.1.0
+version: 1.1.1
```
Comment (alexellis) [inline chart/openfaas/Chart.yaml]: I pushed this up with a namespace change to openfaas/ on the Hub, so it needs a bump to 1.1.2

File: chart/openfaas/README.md [hunk 3]
```diff
@@ -87,6 +87,27 @@ By default services will be exposed with following hostnames (can be changed, se
...
+--from-literal=basic-auth-password=$password 
+```
+
+Update the Helm release and enable basic auth with:
+
+```
+helm upgrade --reuse-values openfaas openfaas/openfaas \
+    --set basic_auth=true
+```
+
```
Comment (alexellis) [inline chart/openfaas/README.md]: Would recommend they do `faas-cli` login somewhere to keep track of the password before losing it.

File: chart/openfaas/Chart.yaml (hunk 2 above)
Comment (stefanprodan) [reply chart/openfaas/Chart.yaml]: This is the chart version, how is related to `openfaas/` Docker Hub?

**Comment by alexellis** [comment]
Sorry about this but my PR created a conflict with the 1.1.1.tgz. Other than that LGTM + some suggestions.

File: chart/openfaas/Chart.yaml (hunk 2 above)
Comment (stefanprodan) [reply chart/openfaas/Chart.yaml]: Ah got it

**Pull request closed without merging**